### ISSUE TITLE
fix(ci): populated:ephemeral-purged leaks 5 hive-mind rows on Ubuntu — launcher purge not running

### ISSUE THREAD

**Issue by eric-cielo** [eric-cielo/moflo#1017]
## Symptom

Consumer-smoke harness on Ubuntu / macOS CI fails with:

```
populated:ephemeral-purged: purgeable rows leaked through #729 purge: hive-mind=N
```

`N` is 1–5 (5 == ROWS_PER_EPHEMERAL_NAMESPACE on Ubuntu, 1 on the latest macOS run). Race-flaky, doesn't reproduce locally on Windows. The companion check `populated:announce-no-legacy-purge-ephemeral-namespace` passes — meaning the launcher's §3e-729 purge had nothing to do at the time it ran. Rows appear AFTER the purge.

## Root cause

The harness runs `flo doctor --json` immediately after the launcher. `doctor-checks-swarm.ts:246` exercises a real `hive-mind_spawn(count: 2) → broadcast → consensus → memory.set/get → shutdown` round-trip. Each step fires `bus.sendUnified(...)` with `namespace: 'hive-mind'` (or `hive-mind-memory`); the `WriteThroughAdapter` listens on `message.unified` and writes those rows to `.moflo/moflo.db` via fire-and-forget `storeEntry`.

`hive-mind_shutdown` is supposed to clean up:

```ts
// src/cli/mcp-tools/hive-mind-tools.ts:847-866
try {
  const adapter = await getWriteThroughAdapter();
  await adapter.clearNamespace(HIVE_NS);          // (1) drain → list → delete
  await adapter.clearNamespace(HIVE_MEMORY_NS);   // (2) same
} catch { ... }
try {
  const bus = await getMessageBus();
  bus.unsubscribe('hive-mind-system');
  if (_writeThroughAdapter) {
    _writeThroughAdapter.detach();                // (3) only here — TOO LATE
    _writeThroughAdapter = null;
  }
} catch { ... }
```

Race: detach happens AFTER clearNamespace. During (1)-(2), the bus is still emitting `message.unified` events from in-flight processing — `terminateAgent` (run earlier in shutdown) emits coordinator messages, the bus's own processing tick (`processingIntervalMs: 50`) drains internal queues, etc. Each event triggers `onUnifiedMessage` → new `storeEntry` call. clearNamespace's `drainPendingWrites()` only catches writes registered *before* it runs; new writes registered between drain and listEntries (or between listEntries and the per-entry deletes) survive.

Cross-platform variance: Linux/macOS bus tick + storeEntry are fast enough to land 1–5 rows in this window. Windows is just slow enough that the same writes happen earlier (before purge) and the launcher's §3e-729 purge cleans them — that's the orthogonal `populated:announce-no-legacy-purge-ephemeral-namespace` failure I saw locally on Windows but not on CI.

## Fix

Detach the adapter BEFORE clearNamespace. With the listener removed, no new bus events can register storeEntry calls during the list+delete loop.

```ts
if (_writeThroughAdapter) {
  _writeThroughAdapter.detach();           // (1) listener gone — no new writes
  try {
    await _writeThroughAdapter.clearNamespace(HIVE_NS);          // (2) drain in-flight + clear
    await _writeThroughAdapter.clearNamespace(HIVE_MEMORY_NS);
  } catch { /* best-effort */ }
  _writeThroughAdapter = null;
}
```

Considered but rejected:
- Add a post-doctor purge in the harness — masks the real bug; doctor's contract is "leaves no side effects".
- Have `clearNamespace` retry list+delete in a loop — papers over the race, doesn't actually close it (writes can still fire during the retry).
- Skip the swarm check in doctor — drops the very tripwire that catches epic #798's regression class.

## Acceptance criteria

- [ ] After `hive-mind_shutdown` returns, `.moflo/moflo.db` has zero rows in the `hive-mind` and `hive-mind-memory` namespaces.
- [ ] The fix is in `src/cli/mcp-tools/hive-mind-tools.ts` — detach moves before clearNamespace, adapter ref captured before null-out.
- [ ] `populated:ephemeral-purged` passes on Ubuntu, macOS, and Windows in consumer-smoke CI.
- [ ] No regression in `populated:announce-no-legacy-purge-ephemeral-namespace` (purge still has nothing to do).
- [ ] Existing `hive-mind-tools` unit tests still pass.
- [ ] `tsc --noEmit` clean; full `npm test` passes (or maintains the same baseline as `main`).

## Test cases

1. **Detach precedes clear** — unit test on `hive-mind_shutdown`: spawn workers, broadcast a message, assert that the adapter is detached BEFORE clearNamespace runs (verified by emitting a bus event mid-shutdown and asserting it does NOT land in the post-shutdown DB).
2. **Post-shutdown DB is clean** — unit test: full hive-mind_init → spawn → broadcast → consensus → memory.set → shutdown round-trip, assert zero rows in `hive-mind` and `hive-mind-memory` namespaces.
3. **Idempotent re-shutdown** — calling shutdown twice doesn't throw and leaves the DB in the same state.

## Consumer impact

Runtime fix in shipped code (`src/cli/mcp-tools/hive-mind-tools.ts`). Used by every consumer that runs `flo doctor` or invokes hive-mind MCP tools. Needs publish + reinstall before consumer smoke is green. Helper API surface unchanged; no state migration.

**Comment by eric-cielo** [comment]
Reopening — the #1024 fix is not 100%. Same `populated:ephemeral-purged hive-mind=1` failure surfaced on PR #1026's first ubuntu-latest CI run (PR #1026 had no hive-mind code changes). Re-run passed, confirming this is an intermittent flake that #1024 narrowed but did not eliminate.

CI evidence:
- PR #1026 first run, ubuntu-latest: FAIL — `purgeable rows leaked through #729 purge: hive-mind=1`
- PR #1026 rerun, ubuntu-latest: PASS

Suspected remaining race: `purgeHiveNamespacesDirect()` opens sql.js directly on the DB file. Per the single-writer architecture (#981 / story #985), if the daemon is alive at the time of hive-mind shutdown its in-memory snapshot can flush AFTER the on-disk DELETE and clobber the purge. The launcher's `purgeEphemeralNamespaces` (#729) avoids this by running BEFORE the daemon starts; `purgeHiveNamespacesDirect` (added in #1024) cannot make that guarantee because hive-mind shutdown happens during normal doctor checks while the daemon is up.

Likely fix: route the bulk delete through the daemon HTTP RPC (`/api/memory/batch` or extend with a namespace-delete route) when the daemon is alive, falling back to raw sql.js only when no daemon is present. That way the daemon's in-memory state stays consistent with disk and cannot clobber. Will pick up next.
